tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py

```diff
@@ -108,6 +108,58 @@ def test_gen_kw(tmpdir, config_str, expected, extra_files, expectation):
             )
 
 
+@pytest.mark.integration_test
+@pytest.mark.parametrize(
+    "config_str, expected, extra_files, expectation, relpath",
+    [
+        (
+            "GEN_KW KW_NAME template.txt somepath/kw.txt prior.txt\nRANDOM_SEED 1234",
+            "MY_KEYWORD -0.881423",
+            [],
+            does_not_raise(),
+            "somepath/",
+        ),
+        # This test was added to show current behaviour for Ert.
+        # If absolute paths should be possible to be used like this is up for debate.
+        (
+            "GEN_KW KW_NAME template.txt /tmp/somepath/kw.txt prior.txt INIT_FILES:custom_param%d",
+            "MY_KEYWORD 1.31",
+            [("custom_param0", "MY_KEYWORD 1.31")],
+            does_not_raise(),
+            "tmp/somepath/",  # removed leading '/' as ert removes this currently
+        ),
+    ],
+)
+def test_gen_kw_outfile_will_use_paths(
+    tmpdir, config_str, expected, extra_files, expectation, relpath
+):
+    with tmpdir.as_cwd():
+        config = dedent(
+            """
+        JOBNAME my_name%d
+        NUM_REALIZATIONS 1
+        """
+        )
+        config += config_str
+        with open("config.ert", mode="w", encoding="utf-8") as fh:
+            fh.writelines(config)
+        with open("template.txt", mode="w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
+        with open("prior.txt", mode="w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD NORMAL 0 1")
+        for fname, contents in extra_files:
+            write_file(fname, contents)
+
+        with expectation:
+            create_runpath("config.ert")
+            assert (
+                Path(
+                    "simulations/realization-0/iter-0/" + relpath + "kw.txt"
+                ).read_text(encoding="utf-8")
+                == expected
+            )
```

Comment:
Maybe something like this?

```suggestion
def test_gen_kw_outfile_will_use_paths(
    tmpdir, relpath
):
    with tmpdir.as_cwd():
        config = dedent(
            f"""
        JOBNAME my_name%d
        NUM_REALIZATIONS 1
        GEN_KW KW_NAME template.txt {relpath} prior.txt
        """
        )
        with open("config.ert", mode="w", encoding="utf-8") as fh:
            fh.writelines(config)
        with open("template.txt", mode="w", encoding="utf-8") as fh:
            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
        with open("prior.txt", mode="w", encoding="utf-8") as fh:
            fh.writelines("MY_KEYWORD NORMAL 0 1")

         create_runpath("config.ert")
         assert (
                Path(
                    f"simulations/realization-0/iter-0/{relpath}kw.txt"
                ).read_text(encoding="utf-8")
                == expected
            )
```

Reply:
Great suggestion, have updated my code somewhat similar.